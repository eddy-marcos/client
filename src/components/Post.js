import React, { useState } from 'react';
import '../styles/post.scss';
import Comments from './Comments';
import NewPost from './NewPost';

const Post = ({ post }) => {
    const [currentPost, setcurrentPost] = useState(null);
    const [postsWithComments, setPostsWithComments] = useState({ comm: 0});
    const [showNewCommentsButton, setShowNewCommentsButton] = useState(false);
    const { id, title, body: postBody } = post;

    const renderComments = () => {
        setcurrentPost(id);
        setPostsWithComments({ ...postsWithComments, comm: id });
        setShowNewCommentsButton(true);
    }

    return (
        <>
            <div className="post" data-testid = "post-element">
                <h3>{title}</h3>
                <p>{postBody}</p>
                <button className="comment-button" onClick={renderComments}>Show Comments</button>

                <div>
                    {
                    postsWithComments ? 
                        <>
                            <Comments currentPost={currentPost}/> 
                            <NewPost id={id} showNewCommentsButton={showNewCommentsButton}/>
                        </>
                    : '' }

                 </div>
            </div>
        </>
    )
}

export default Post
