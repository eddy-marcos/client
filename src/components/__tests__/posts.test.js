import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Post from '../Post';

afterEach( () => {
    cleanup();
});

test('post should contain data', () => {
   
    const post = {
        "userId": 1,
        "id": 1,
        "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
        "body": "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
     }
    
    render(<Post post={post}/>);

    const postElement = screen.getByTestId('post-element');

    expect(postElement).toBeVisible();    
    expect(postElement).toBeValid();
    expect(postElement).toContainHTML('<div></div>');
    expect(postElement).toBeTruthy();
})

test('post component matches snapshot', () => {
    const post = {
        "userId": 1,
        "id": 1,
        "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
        "body": "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
     }

     const tree = renderer.create(<Post post={post}/>).toJSON();
     expect(tree).toMatchSnapshot();
})