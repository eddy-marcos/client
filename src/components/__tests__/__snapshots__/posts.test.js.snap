// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post component matches snapshot 1`] = `
<div
  className="post"
  data-testid="post-element"
>
  <h3>
    sunt aut facere repellat provident occaecati excepturi optio reprehenderit
  </h3>
  <p>
    quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
  </p>
  <button
    className="comment-button"
    onClick={[Function]}
  >
    Show Comments
  </button>
  <div>
    
  </div>
</div>
`;
